frontend/app/components/Spots/SpotPlayer/components/AccessModal.tsx

```diff
@@ -35,70 +36,64 @@ function AccessModal() {
     spotStore.pubKey ? `?pub_key=${spotStore.pubKey.value}` : ''
   }`;
 
+  useEffect(() => {
+    if (spotStore.pubKey) {
+      const interval = Object.keys(expirationValues).find(key => expirationValues[key as Intervals] === spotStore.pubKey.expiration);
+      if (interval) setSelectedInterval(Number(interval) as Intervals);
+    }
+  }, [spotStore.pubKey]);
+
   const menuItems = [
     {
-      key: Intervals.hour,
+      key: Intervals.hour.toString(),
```

Comment:
?

Reply:
The one we earlier had counts everything in numbers. E.g. if a link is expiring in 7 days, it says Link Expiring in 168 hrs. So that was fixed.